libs/vkd3d/state.c

```diff
@@ -4502,6 +4710,20 @@ static HRESULT d3d12_pipeline_state_init_graphics_create_info(struct d3d12_pipel
         ++graphics->stage_count;
     }
 
+    if (state->pipeline_type == VKD3D_PIPELINE_TYPE_GRAPHICS)
```

Comment:
Couldn't this be validated in-line in the per-stage loop?

Reply:
Fair, moved it to where we parse the PC input signature.